### PR TITLE
/email/first_seen, /device/session_age and /device/session_id

### PR DESCRIPTION
**Pull request by oalders**

### CODE REVIEW

File: src/MinFraud/Validation/Rules/Device.php [hunk 1]
```diff
@@ -15,6 +15,8 @@ public function __construct()
         $this->validatable = v::keySet(
             v::key('accept_language', v::stringType(), false),
             v::key('ip_address', v::ip(), true),
+            v::key('session_id', v::stringType(), false),
+            v::key('session_age', v::floatType(), false),
```
Comment (oschwald) [inline src/MinFraud/Validation/Rules/Device.php]: `v::floatType()->min(0, true)` might be nice here

File: src/MinFraud/Validation/Rules/Device.php [hunk 2]
```diff
@@ -15,6 +15,8 @@ public function __construct()
         $this->validatable = v::keySet(
             v::key('accept_language', v::stringType(), false),
             v::key('ip_address', v::ip(), true),
+            v::key('session_id', v::stringType(), false),
```
Comment (oschwald) [inline src/MinFraud/Validation/Rules/Device.php]: Maybe ` v::stringType()->length(1, 255)`

File: README.md [hunk 3]
```diff
@@ -135,8 +135,10 @@ use MaxMind\MinFraud;
 $mf = new MinFraud(1, 'ABCD567890');
```
Comment (oschwald) [inline README.md]: You didn't update `CHANGELOG.md`

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,8 @@ CHANGELOG
   `ebs`, `hipay`, and `lemon_way`.
 * Updated the docs for `MaxMind\MinFraud\Model\Address` now that
   `isPostalInCity` may be returned for addresses world-wide.
+* The `first_seen` output was added to the `Email` response model.
+* `session_age` and `session_id` inputs were added to `Email` model.
```
Comment (oschwald) [inline CHANGELOG.md]: "were added to `Email` model" is wrong.

File: src/MinFraud/Validation/Rules/Device.php [hunk 5]
```diff
@@ -12,9 +12,12 @@ class Device extends AbstractWrapper
 {
     public function __construct()
     {
+        $int_or_string = new IntOrString();
```
Comment (oschwald) [inline src/MinFraud/Validation/Rules/Device.php]: Is this used?

File: tests/MaxMind/Test/MinFraudTest.php [hunk 6]
```diff
@@ -483,6 +482,105 @@ public function badIps()
...
+        $this->createMinFraudRequestWithFullResponse(
+            'insights',
+            0
+        )->withDevice(['ip_address' => '1.2.3.4', 'session_age' => $age]);
+    }
+
+    public function badSessionAge()
+    {
+        return [
+            ["X"],
```
Comment (oschwald) [inline tests/MaxMind/Test/MinFraudTest.php]: Maybe add a negative value here.

File: tests/MaxMind/Test/MinFraudTest.php [hunk 7]
```diff
@@ -483,6 +482,105 @@ public function badIps()
...
+            'insights',
+            0
+        )->withDevice(['ip_address' => '1.2.3.4', 'session_id' => $id]);
+    }
+
+    public function goodSessionId()
+    {
+        return [
+            [0],
+            [3600],
```
Comment (oschwald) [inline tests/MaxMind/Test/MinFraudTest.php]: Add a stringy test ID here.

File: tests/MaxMind/Test/MinFraudTest.php (hunk 6 above)
Comment (oalders) [inline tests/MaxMind/Test/MinFraudTest.php]: I'm doing that in `testNegativeSessionAge`.  Is that an appropriate place to do it?

File: CHANGELOG.md (hunk 4 above)
Comment (oalders) [inline CHANGELOG.md]: Ugh.  I've just fixed up the initial commit.

File: src/MinFraud/Validation/Rules/Device.php (hunk 5 above)
Comment (oalders) [inline src/MinFraud/Validation/Rules/Device.php]: Leftover debugging.  Fixed via 48e257ad7

File: tests/MaxMind/Test/MinFraudTest.php (hunk 7 above)
Comment (oalders) [inline tests/MaxMind/Test/MinFraudTest.php]: Added in 43bb7147fa27

File: tests/MaxMind/Test/MinFraudTest.php (hunk 6 above)
Comment (oschwald) [inline tests/MaxMind/Test/MinFraudTest.php]: Ah, ok. I missed that. It seems fine to do it here.

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,6 +5,8 @@ CHANGELOG
   `ebs`, `hipay`, and `lemon_way`.
 * Updated the docs for `MaxMind\MinFraud\Model\Address` now that
   `isPostalInCity` may be returned for addresses world-wide.
+* The `first_seen` output was added to the `Email` response model.
+* `session_age` and `session_id` inputs were added to `Device` model.
```
Comment (oschwald) [inline CHANGELOG.md]: There is no `Device` model. You added these to the `device` input validation.

Comment (oalders) [inline CHANGELOG.md]: :)  I've pushed another fixup.

**Pull request merged**